Warning at compilation

Hello,

For information, I use `gcc (Ubuntu 9.2.1-17ubuntu1~18.04.1) 9.2.1 20191102` and I got the following 2 warning when I compile the current [master ](https://github.com/getsentry/sentry-native/commit/8a38e5788248a42ede72e8504fb730046f6bde60) branch:

```
/libsentry/src/sentry_json.c: In function ‘sentry__jsonwriter_write_double’:
/libsentry/src/sentry_json.c:228:21: warning: comparison of unsigned expression < 0 is always false [-Wtype-limits]
  228 |         if (written < 0 || written >= sizeof(buf) || !isfinite(val)) {
      |                     ^
/libsentry/src/sentry_json.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-gnu-include-next’
```

Regards

--- a/src/sentry_json.c
+++ b/src/sentry_json.c
@@ -221,11 +221,10 @@ sentry__jsonwriter_write_double(sentry_jsonwriter_t *jw, double val)
     if (can_write_item(jw)) {
         char buf[50];
         // The MAX_SAFE_INTEGER is 9007199254740991, which has 16 digits
-        size_t written
-            = (size_t)sentry__snprintf_c(buf, sizeof(buf), "%.16g", val);
+        int written = sentry__snprintf_c(buf, sizeof(buf), "%.16g", val);
         // print `null` if we have printf issues or a non-finite double, which
         // can't be represented in JSON.
-        if (written < 0 || written >= sizeof(buf) || !isfinite(val)) {
+        if (written < 0 || written >= (int)sizeof(buf) || !isfinite(val)) {
             write_str(jw, "null");
         } else {
             buf[written] = '\0';

